### PR TITLE
Fixed a bug where events wouldn't be discarded if no stack id was provided

### PR DESCRIPTION
**Pull request by niemyjski**

### CODE REVIEW

File: tests/Exceptionless.Tests/Pipeline/EventPipelineTests.cs [hunk 1]
```diff
@@ -843,6 +843,43 @@ public sealed class EventPipelineTests : IntegrationTestsBase {
...
+            await RefreshDataAsync();
+            
+            ev = EventData.GenerateEvent(organizationId: TestConstants.OrganizationId, projectId: TestConstants.ProjectId, type: Event.KnownTypes.Log, source: "test", occurrenceDate: SystemClock.OffsetNow);
+            context = await _pipeline.RunAsync(ev, organization, project);
+            Assert.False(context.IsProcessed);
+            Assert.False(context.HasError);
+            Assert.True(context.IsCancelled);
+            Assert.True(context.IsDiscarded);
+            await RefreshDataAsync();
+        }
```
Comment (ejsmith) [inline tests/Exceptionless.Tests/Pipeline/EventPipelineTests.cs:882]: Can we do some asserts to make sure it didn't count against your plan limits?

File: tests/Exceptionless.Tests/Pipeline/EventPipelineTests.cs [hunk 2]
```diff
@@ -843,6 +843,43 @@ public sealed class EventPipelineTests : IntegrationTestsBase {
                 Assert.Null(userDescription?.EmailAddress);
             }
         }
+        
+        [Fact]
+        public async Task WillHandleDiscardedStack() {
+            var organization = OrganizationData.GenerateSampleOrganization(_billingManager, _plans);
+            var project = ProjectData.GenerateSampleProject();
```
Comment (ejsmith) [inline tests/Exceptionless.Tests/Pipeline/EventPipelineTests.cs:850]: Why not use our new event builder to create an event?

File: src/Exceptionless.Core/Pipeline/010_AssignToStackAction.cs [hunk 3]
```diff
@@ -109,11 +109,12 @@ public class AssignToStackAction : EventPipelineActionBase {
...
-                        ctx.IsDiscarded = true;
-                        ctx.IsCancelled = true;
-                    }
+                }
+                
+                if (ctx.Stack.Status == StackStatus.Discarded) {
+                    ctx.IsDiscarded = true;
+                    ctx.IsCancelled = true;
+                    continue;
                 }
```
Comment (ejsmith) [inline src/Exceptionless.Core/Pipeline/010_AssignToStackAction.cs:118]: This seems good.

**Review by ejsmith — CHANGES_REQUESTED** [review]

File: tests/Exceptionless.Tests/Pipeline/EventPipelineTests.cs (hunk 2 above)
Comment (niemyjski) [reply tests/Exceptionless.Tests/Pipeline/EventPipelineTests.cs:850]: No, that doesn't go through the pipeline.

File: tests/Exceptionless.Tests/Pipeline/EventPipelineTests.cs (hunk 1 above)
Comment (niemyjski) [reply tests/Exceptionless.Tests/Pipeline/EventPipelineTests.cs:882]: We can't here, that happens in the job not in the pipeline.

**Review by ejsmith — APPROVED** [review]

**Pull request merged**